azure/services/managedclusters/spec.go

```diff
@@ -165,7 +166,7 @@ type LoadBalancerProfile struct {
 // APIServerAccessProfile is the access profile for AKS API server.
 type APIServerAccessProfile struct {
 	// AuthorizedIPRanges are the authorized IP Ranges to kubernetes API server.
-	AuthorizedIPRanges []string
+	AuthorizedIPRanges *[]string
```

Comment:
Like Cecile said, could you explain why we're changing this into a `*[]string`?

Reply:
The original reason for this change was for 
1. consistency with Azure SDK
2. CAPZ can differentiate between unset field vs empty array. 

But since a slice also has a nil type, I have reverted this to `[]string` type instead of a pointer.